src/lib/MovingAverage.java

```diff
@@ -0,0 +1,36 @@
+package lib;
+
+public class MovingAverage {
```

Comment:
There is a more efficient implementation that uses a queue.  This implementation caches the current sum.  It adds a new value to the queue until it reaches the desired number of samples.  Once `numSamples` is reached, it pops a value before adding each new one and subtracts it from the cached total.  The new value is then added to the cached sum and pushed onto the queue.  When the result is computed, the number is the minimum of numSamples and the queue's length.